pkg/scalers/aws_dynamodb_scaler.go

```diff
@@ -0,0 +1,249 @@
+		names, err := json2Map(val)
+
+		if err != nil {
+			return nil, fmt.Errorf("error parsing expressionAttributeNames: %s", err)
+		}
+
+		meta.expressionAttributeNames = names
+	} else {
+		return nil, fmt.Errorf("no expressionAttributeNames given")
+	}
+
+	if val, ok := config.TriggerMetadata["expressionAttributeValues"]; ok && val != "" {
+		values, err := json2DynamoMap(val)
+
+		if err != nil {
+			return nil, fmt.Errorf("error parsing expressionAttributeValues: %s", err)
+		}
+
+		meta.expressionAttributeValues = values
+	} else {
+		return nil, fmt.Errorf("no expressionAttributeValues given")
+	}
+
+	if val, ok := config.TriggerMetadata["targetValue"]; ok && val != "" {
+		n, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing metadata targetValue")
+		}
+
+		meta.targetValue = n
+	} else {
+		return nil, fmt.Errorf("no targetValue given")
+	}
+
+	auth, err := getAwsAuthorization(config.AuthParams, config.TriggerMetadata, config.ResolvedEnv)
+	if err != nil {
+		return nil, err
+	}
+
+	meta.awsAuthorization = auth
+	meta.scalerIndex = config.ScalerIndex
+
+	return &meta, nil
+}
+
+func createDynamoDBClient(meta *awsDynamoDBMetadata) *dynamodb.DynamoDB {
+	sess := session.Must(session.NewSession(&aws.Config{
+		Region: aws.String(meta.awsRegion),
+	}))
+
+	var dbClient *dynamodb.DynamoDB
+
+	if !meta.awsAuthorization.podIdentityOwner {
+		dbClient = dynamodb.New(sess, &aws.Config{
+			Region: aws.String(meta.awsRegion),
+		})
+
+		return dbClient
+	}
+
+	creds := credentials.NewStaticCredentials(meta.awsAuthorization.awsAccessKeyID, meta.awsAuthorization.awsSecretAccessKey, "")
+
+	if meta.awsAuthorization.awsRoleArn != "" {
+		creds = stscreds.NewCredentials(sess, meta.awsAuthorization.awsRoleArn)
+	}
+
+	dbClient = dynamodb.New(sess, &aws.Config{
+		Region:      aws.String(meta.awsRegion),
+		Credentials: creds,
+	})
+
+	return dbClient
+}
+
+func (c *awsDynamoDBScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	metricValue, err := c.GetQueryMetrics()
+	if err != nil {
+		dynamoDBLog.Error(err, "Error getting metric value")
+		return []external_metrics.ExternalMetricValue{}, err
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *resource.NewQuantity(metricValue, resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+func (c *awsDynamoDBScaler) GetMetricSpecForScaling(context.Context) []v2beta2.MetricSpec {
+	targetMetricValue := resource.NewQuantity(int64(c.metadata.targetValue), resource.DecimalSI)
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: GenerateMetricNameWithIndex(c.metadata.scalerIndex, kedautil.NormalizeString(fmt.Sprintf("aws-dynamodb-%s", c.metadata.tableName))),
```

Comment:
Could you please move this to `parseAwsDynamoDBMetadata()` so it is called only on the first time?

You can take a look here, to see the example:
https://github.com/kedacore/keda/blob/b6bc7b754e2b5145c45895126e9dc57e86ab8816/pkg/scalers/datadog_scaler.go#L134